### ISSUE TITLE
perhaps wrong error thrown

### ISSUE THREAD

**Issue by lmiq** [MilesCranmer/BorrowChecker.jl#41]
I cannot understand how this is throwing an error: 

```julia
julia> using BorrowChecker, StaticArrays

julia> BorrowChecker.@auto scope=:function function f(x)
           l = 1
           return SVector{l, Int}(x)
       end
f (generic function with 1 method)

julia> f([1])
ERROR: BorrowCheckError for specialization Tuple{typeof(f), Vector{Int64}}

  method: f(x) @ Main REPL[23]:1

  [1] stmt#12: cannot perform consume: value is aliased by another live binding      at REPL[23]:3
        1     BorrowChecker.@auto scope=:function function f(x)
        2         l = 1
      > 3         return SVector{l, Int}(x)
        4     end

      stmt: (%11)(_2)

  [2] stmt#12: value escapes/consumed by unknown call; it (or an alias) is used later      at REPL[23]:3
        1     BorrowChecker.@auto scope=:function function f(x)
        2         l = 1
      > 3         return SVector{l, Int}(x)
        4     end

      stmt: (%11)(_2)
Stacktrace:
 [1] (::BorrowChecker.Auto.var"#44#45"{BorrowChecker.Auto.Config, UInt64, Type{Tuple{typeof(f), Vector{Int64}}}})()
   @ BorrowChecker.Auto ~/.julia/packages/BorrowChecker/2Siwv/src/auto/frontend.jl:202
 [2] _with_reflection_ctx(f::Function, world::UInt64)
   @ BorrowChecker.Auto ~/.julia/packages/BorrowChecker/2Siwv/src/auto/summaries.jl:33
 [3] check_signature(tt::Type{<:Tuple}; cfg::BorrowChecker.Auto.Config, world::UInt64)
   @ BorrowChecker.Auto ~/.julia/packages/BorrowChecker/2Siwv/src/auto/frontend.jl:193
 [4] check_signature
   @ ~/.julia/packages/BorrowChecker/2Siwv/src/auto/frontend.jl:188 [inlined]
 [5] _generated_assert_safe_body(world::UInt64, lnn::Method, this::Type, sig::Type)
   @ BorrowChecker.Auto ~/.julia/packages/BorrowChecker/2Siwv/src/auto/generated.jl:67
 [6] f(x::Vector{Int64})
   @ Main ./REPL[23]:1
 [7] top-level scope
   @ REPL[24]:1

julia> BorrowChecker.@auto scope=:function function f(x)
           return SVector{1, Int}(x)
       end
f (generic function with 1 method)

julia> f([1])
1-element SVector{1, Int64} with indices SOneTo(1):
 1
````

**Comment by MilesCranmer** [comment]
Thanks for the report! I'll figure this out and fix
